src/include/migraphx/operation.hpp

```diff
@@ -733,12 +782,14 @@ struct operation
         virtual bool is_context_free() const                                       = 0;
         virtual bool need_normalization() const                                    = 0;
         virtual bool has_finalize() const                                          = 0;
+        virtual bool has_offload_copy() const                                      = 0;
         virtual lifetime get_lifetime() const                                      = 0;
         virtual std::ptrdiff_t output_alias(const std::vector<shape>& input) const = 0;
         virtual value
         compile(context& ctx, const shape& output, const std::vector<shape>& input) = 0;
         virtual void
         finalize(context& ctx, const shape& output, const std::vector<shape>& input) = 0;
+        virtual bool runs_on_offload_target() const                                  = 0;
```

Comment:
These methods should not be added to the operation class. They should be set as attributes.

Reply:
I need a way to distinguish between regular MIGraphX op and custom op. If i just have an attribute, i need some other mechanism to classify regular MIGrpahX op v/s custom op.

Reply:
I dont think we need to make a distinction. Either way, you can add a `custom_op` attribute.

Reply:
Added logic to use `custom_op` attribute and `target` attribute.